lib/sda/host.go

```diff
@@ -381,6 +387,20 @@ func (h *Host) processMessages() {
 	}
 }
 
+func (h *Host) processRequest(e *network.Entity, r *Request) {
+	// check if the target service is indeed existing
+	var s Service
+	var ok bool
+	if s, ok = h.serviceStore.serviceByID(r.Service); !ok {
```

Comment:
This is why I put it in the line above. Then you have it available for later. I can live with the fact that you want to write concise `if`-statements, but if the variables are used outside of the scope of the if, it's bizarre to define the variables beforehand, when you can do it with a `:=`.